examples/docker2singularity/docker2singularity.sh

```diff
@@ -85,37 +93,72 @@ echo "Size: $size MB for the singularity container"
 ################################################################################
 ### IMAGE CREATION #############################################################
 ################################################################################
-
+TMPDIR=$(mktemp -u -d)
+mkdir -p $TMPDIR
 
 creation_date=`echo ${creation_date} | cut -c1-10`
 new_container_name=$image_name-$creation_date.img
 $SUDOCMD singularity create -s $size $new_container_name
 $SUDOCMD docker export $container_id | $SUDOCMD singularity import $new_container_name
-$SUDOCMD docker inspect $container_id >> singularity.json
-sudo singularity copy $new_container_name singularity.json /
+$SUDOCMD docker inspect $container_id >> $TMPDIR/singularity.json
+sudo singularity copy $new_container_name $TMPDIR/singularity.json /
 
 # Merge the /etc/group file
-$SUDOCMD docker cp $container_id:/etc/group grouphost
-sort /etc/group grouphost | uniq -u > group
-$SUDOCMD singularity copy $new_container_name group /etc/group
-rm singularity.json
-$SUDOCMD rm grouphost
-$SUDOCMD rm group
+$SUDOCMD docker cp $container_id:/etc/group $TMPDIR/grouphost
+sort /etc/group $TMPDIR/grouphost | uniq -u > $TMPDIR/group
+$SUDOCMD singularity copy $new_container_name $TMPDIR/group /etc/group
+# Bootstrap the image to set up scripts for environemnt setup
+$SUDOCMD singularity bootstrap $new_container_name
+$SUDOCMD chmod a+rw -R $TMPDIR
 
 ################################################################################
 ### SINGULARITY RUN SCRIPT #####################################################
 ################################################################################
 
 CMD=$($SUDOCMD docker inspect --format='{{json .Config.Cmd}}' $image)
-# Remove quotes and braces
-CMD=`echo "${CMD//\"/}" | sed 's/\[//g' | sed 's/\]//g'`
-if [[ $CMD != none ]]; then
-  echo '#!/bin/sh'
-  (IFS='[],'; echo $CMD)
-fi > singularity
-chmod +x singularity
-sudo singularity copy $new_container_name singularity /
-rm singularity
+if [[ $CMD != [* ]]; then
+	CMD="/bin/sh -c "$CMD
+fi
+# Remove quotes, commas, and braces
+CMD=`echo "${CMD//\"/}" | sed 's/\[//g' | sed 's/\]//g' | sed 's/,//g'`
+
+ENTRYPOINT=$($SUDOCMD docker inspect --format='{{json .Config.Entrypoint}}' $image)
+if [[ $ENTRYPOINT != [* ]]; then
+	ENTRYPOINT="/bin/sh -c "$ENTRYPOINT
+fi
+# Remove quotes, commas, and braces
+ENTRYPOINT=`echo "${ENTRYPOINT//\"/}" | sed 's/\[//g' | sed 's/\]//g' | sed 's/,//g'`
+
+echo '#!/bin/sh' > $TMPDIR/singularity
+if [[ $ENTRYPOINT != "null" ]]; then
+    echo $ENTRYPOINT '$@' >> $TMPDIR/singularity;
+else
+    echo $CMD '$@' >> $TMPDIR/singularity;
+fi
+
```

Comment:
For the entrypoint, testing with `ubuntu:latest` I get:

      #!/bin/sh
      /bin/sh -c null $@

then when I try to run the container:

      /bin/sh: 1: null: not found

Maybe if there isn't a run command, we should just not add a runscript? Do you think there is reason to try to enforce a "null" kind of runscript in containers?